### PR TITLE
Update Return error macro

### PR DESCRIPTION
**Pull request by parthishere**
Fixing RETURN_ERROR macro. 
It is causing problem in Coverity Static analysis tool as we are directly converting the UINT value to INTN

Changing value from UINT to INTN might cause problem. Here we know that the values would not be in loss of data. To increase the code quality and increase the static tool analysis score we have to change it

Cc: Jiangang He <jiangang.he@amd.com>
Cc: Neo Hsueh <Hong-Chih.Hsueh@amd.com>

# Description

It is causing problem in Coverity Static analysis tool as we are directly converting the UINT value to INTN

## How This Was Tested

Tested by using edge case values and unit test cases

### CODE REVIEW

File: MdePkg/Include/Base.h [hunk 1]
```diff
@@ -1058,7 +1058,7 @@ typedef UINTN RETURN_STATUS;
   @retval FALSE         The high bit of StatusCode is clear.
 
 **/
-#define RETURN_ERROR(StatusCode)  (((INTN)(RETURN_STATUS)(StatusCode)) < 0)
+#define RETURN_ERROR(StatusCode)  (((RETURN_STATUS)(StatusCode)) > MAX_BIT)
```
[the remark this answers is not in the document]
Comment (parthishere) [reply MdePkg/Include/Base.h:1061]: 1) Comparison operator:
using '>' instead of '>=' is more accurate. Here's why:


We're encoding errors as MAX_BIT | StatusCode
This means other bits besides the most significant one could be set
'>' ensures we catch all error cases, not just the minimum error value
2) Edge case consideration:

While 1000 0000 is technically a negative number in signed context, in our unsigned error encoding scheme, it represents the minimum error value. Using '<' excludes this value, which is probably what we want - to differentiate between the highest success code and the lowest error code.

[the remark this answers is not in the document]
Comment (mdkinney) [reply MdePkg/Include/Base.h:1061]: I disagree.  If a change is being made here for static analysis, then we do not want to change the behavior of this macro.  A request to change the behavior of this macro should be a different PR.

[the remark this answers is not in the document]
Comment (parthishere) [reply MdePkg/Include/Base.h:1061]: Understood, I will update the macro without changing any logic and will create a new PR if needed.

**Comment by lgao4** [comment]
please merge commit to single one, and follow https://github.com/tianocore/tianocore.github.io/wiki/Commit-Message-Format to   formalize the commit message

**Comment by parthishere** [comment]
I have squashed the commits, please check the commit

**Review by mdkinney — APPROVED** [review]

**Comment by lgao4** [comment]
> I have squashed the commits, please check the commit

The commit message format should be refined based on the above commit message format wiki.

**Comment by lgao4** [comment]
This is not critical fix. I think it can be merged after the stable tag 202408.

**Review by lgao4 — APPROVED** [review]

**Comment by mdkinney** [comment]
I have asked Mergify support to determine why this PR is not being merged.

**Comment by mdkinney** [comment]
@mergifyio requeue
